### PR TITLE
Introduce a default value mechanism for validAfterBlockNumber

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
Instead of using -1 as a literal value this introduces a mechanism in which -1 validAfterBlockNumber means "pick up the max block number you know" and use it.

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
no ticket.

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try-

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try-

**Comment by dzajkowski** [comment]
bors try

File: casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala [hunk 1]
```diff
@@ -176,14 +176,34 @@ class MultiParentCasperImpl[F[_]: Sync: Concurrent: Sync: ConnectionsCell: Trans
...
+      def updateDeployValidAfterBlock(deployData: DeployData, max: Long) =
+        if (deployData.validAfterBlockNumber == -1)
+          deployData.withValidAfterBlockNumber(max)
+        else
+          deployData
+
+      def updateDeployHistory(state: CasperState, max: Long) =
+        state.copy(deployHistory = state.deployHistory.map(deployData => {
+          updateDeployValidAfterBlock(deployData, max)
+        }))
```
Comment (ArturGajowy) [inline casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala]: looks like a good use case for lenses :zipper_mouth_face:

File: casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala [hunk 2]
```diff
@@ -176,14 +176,34 @@ class MultiParentCasperImpl[F[_]: Sync: Concurrent: Sync: ConnectionsCell: Trans
...
+        maxBlockNumber = ProtoUtil.maxBlockNumber(p)
+        /*
+         * This mechanism is a first effort to make life of a deploying party easier.
+         * Instead of expecting the user to guess the current block number we assume that
+         * if no value is given (default: -1) rchain should try to deploy
+         * with the current known max block number.
+         *
+         * TODO make more developer friendly by introducing Option instead of a magic number
+         */
+        _                <- Cell[F, CasperState].modify(state => updateDeployHistory(state, maxBlockNumber))
```
Comment (ArturGajowy) [inline casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala]: I've got very superficial non-understanding of the issue at hand, but wouldn't having a default offset from the given node's known maxBlockNumber make things better?

If I give a value bigger than my known max block - I'm proposing an invalid block according to my knowledge?

If I give a value bigger than other node's max block number - will they consider the block invalid until they get it from the other node? Maybe I should propose a smaller number?

**Pull request merged**